src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexingService.cs

```diff
@@ -53,193 +61,172 @@ public class ElasticIndexingService
             _indexingTaskManager = indexingTaskManager;
             _elasticConnectionOptions = elasticConnectionOptions.Value;
             _siteService = siteService;
+            _contentManager = contentManager;
+            _contentItemIndexHandlers = contentItemIndexHandlers;
+            _store = store;
             _contentDefinitionManager = contentDefinitionManager;
             _logger = logger;
         }
 
-        public async Task ProcessContentItemsAsync(string indexName = default)
+        public async Task ProcessContentItemsAsync(params string[] indexNames)
         {
             if (!_elasticConnectionOptions.FileConfigurationExists())
             {
                 return;
             }
 
+            var indexSettingsList = await _elasticIndexSettingsService.GetSettingsAsync();
+
+            if (indexNames != null && indexNames.Length > 0)
+            {
+                indexSettingsList = indexSettingsList
+                    .Where(x => indexNames.Contains(x.IndexName, StringComparer.OrdinalIgnoreCase));
+            }
+
+            if (!indexSettingsList.Any())
+            {
+                return;
+            }
+
             var allIndices = new Dictionary<string, long>();
             var lastTaskId = long.MaxValue;
-            IEnumerable<ElasticIndexSettings> indexSettingsList = null;
 
-            if (string.IsNullOrEmpty(indexName))
+            // Find the lowest task id to process.
+            foreach (var indexSetting in indexSettingsList)
             {
-                indexSettingsList = await _elasticIndexSettingsService.GetSettingsAsync();
+                var taskId = await _indexManager.GetLastTaskId(indexSetting.IndexName);
+                lastTaskId = Math.Min(lastTaskId, taskId);
+                allIndices.Add(indexSetting.IndexName, taskId);
+            }
 
-                if (!indexSettingsList.Any())
-                {
-                    return;
-                }
+            IEnumerable<IndexingTask> tasks = [];
+            var allContentTypes = indexSettingsList.SelectMany(x => x.IndexedContentTypes ?? []).Distinct().ToArray();
```

Comment:
I'm looking at this again and here you concat all the contentTypes from all index settings to define which ones to index but it is a setting that should be different from one index to an other. Unless you make a distinction per index later on I don't think it will work.

Reply:
Here we get content types that could potentially be indexed.